KsApi/MockService.swift

```diff
@@ -924,24 +932,125 @@
       return SignalProducer(value: self.fetchProjectNotificationsResponse)
     }
 
-    internal func fetchProject(param: Param) -> SignalProducer<Project, ErrorEnvelope> {
+    internal func fetchProject(param _: Param) -> SignalProducer<Project, ErrorEnvelope> {
+      guard let result = self.fetchProjectEnvelopeResult else {
+        return .empty
+      }
+
+      switch result {
+      case let .success(project):
+        return SignalProducer(value: project)
+      case let .failure(error):
+        return SignalProducer(error: error)
+      }
+    }
+
+    internal func fetchProject(projectParam: Param)
+      -> SignalProducer<Project.ProjectPamphletData, ErrorEnvelope> {
       guard let client = self.apolloClient else {
         return .empty
       }
 
-      switch (param.id, param.slug) {
+      switch (projectParam.id, projectParam.slug) {
       case let (.some(paramId), _):
-        let fetchProjectCommentsQuery = GraphAPI
+        let fetchProjectQuery = GraphAPI
           .FetchProjectByIdQuery(projectId: paramId, withStoredCards: false)
 
-        return client
-          .fetchWithResult(query: fetchProjectCommentsQuery, result: self.fetchProjectEnvelopeResult)
+        let projectOrErrorOnlyResult: Result<Project, ErrorEnvelope>
+
+        switch self.fetchProjectPamphletEnvelopeResult {
+        case let .success(projectPamphletData):
+          projectOrErrorOnlyResult = .success(projectPamphletData.project)
+        case let .failure(errorEnvelope):
+          projectOrErrorOnlyResult = .failure(errorEnvelope)
+        case .none:
+          return .empty
+        }
+
+        /**
+         FIXME: Separately attaching passed in `backingId` from `Project.ProjectPamphletData` and not calling the mock client directly with `Project.ProjectPamphletData` because it is not `Decodable` (error, unsure how to correct at this time, because `Project` which is not decodable can be passed in on its' own)
```

Comment:
If making this Decodable a large effort? Not entirely familiar with this particular piece of the app so I can't comment here.

Reply:
In all likelihood I could probably just add `:Decodable` to `Project` object and it would compile, but because we are moving this model over to newer GQL version anyway I've started adding `Decodable` to that model as it will be clear in the next PR that adds the story, risks, campaign and faq. Once every property in this model gets put inside the new `GraphQLProject` model (which is right now a property on `Project`), then the entire model becomes decodable without the worry of needed to test all use-cases of `Project` upfront.